cmapi/integration_tests/conftest.py

```diff
@@ -0,0 +1,183 @@
+    )
+    parser.addoption(
+        "--create-cluster",
+        action="store_true",
+        default=False,
+        help="Create the cluster before running tests",
+    )
+    parser.addoption(
+        "--destroy-cluster",
+        action="store_true",
+        default=False,
+        help="Destroy the cluster after running tests",
+    )
+    parser.addoption(
+        "--ssh-user",
+        default="ubuntu",
+        help="SSH user to connect to the cluster hosts",
+    )
+
+
+@pytest.fixture(scope="session")
+def terraform_dir(request) -> Path:
+    """
+    Fixture to provide the terraform directory path.
+    Fails explicitly if --terraform-dir option is not provided.
+    """
+    terraform_directory = request.config.getoption("--terraform-dir")
+    if not terraform_directory:
+        pytest.fail("--terraform-dir option is required but not provided")
+    terraform_directory = Path(terraform_directory).resolve()
+    if not terraform_directory.is_dir():
+        pytest.fail(f"Specified terraform directory '{terraform_directory}' does not exist")
+    return terraform_directory
+
+
+@pytest.fixture(scope="session")
+def prepared_cluster(request, terraform_dir):
+    """
+    Provides existence of the cluster
+    If it doesn't exist and --create-cluster is passed, it creates it.
+    If it exists and --create-cluster is passed, it re-creates it.
+    If --destroy-cluster is passed, it destroys the cluster when the tests finish.
+    """
+    create_cluster = request.config.getoption("--create-cluster")
+    destroy_cluster = request.config.getoption("--destroy-cluster")
+
+    cluster_already_exists = cluster_mgmt.cluster_exists(terraform_dir)
+
+    if create_cluster:
+        if cluster_already_exists:
+            print("Cluster already exists, destroying it first...")
+            cluster_mgmt.destroy_cluster(terraform_dir)
+        print("Creating the cluster...")
+        cluster_mgmt.create_cluster(terraform_dir)
+
+
+    yield
+
+
+    if destroy_cluster:
+        print("Destroying the cluster...")
+        cluster_mgmt.destroy_cluster(terraform_dir)
+
+
+@pytest.fixture(scope="session")
+def cluster_config(request, terraform_dir, prepared_cluster) -> ClusterConfig:
+    """
+    Fixture to provide the cluster configuration.
+    """
+    with change_directory(terraform_dir):
+        # Read the cluster configuration from the terraform output
+        outputs = json.loads(subprocess.check_output(
+            ["terraform", "output", "-json"],
+            text=True,
+        ))
+
+        if outputs == {}:
+            pytest.fail("Terraform output is empty, looks like cluster is not yet created. Pass --create-cluster option to create it.")
+
+        key_file_path = Path(outputs["ssh_key_file"]["value"])
+        ssh_user = request.config.getoption("--ssh-user")
+
+        mcs_nodes = []
+        for host_descr in outputs["columnstore_nodes"]["value"]:
```

Comment:
Is it safe here? Are 100% sure that this key exist? Get looks better in this case.